packages/ai/src/AINoticeIndicatorTemplate.tsx

```diff
@@ -14,29 +14,38 @@ export default function AINoticeIndicatorTemplate(this: AINoticeIndicator) {
 				role="button"
 				aria-haspopup="dialog"
 				aria-expanded="true"
-				aria-label="string">
-				{this.isIconOnly ? 
+				aria-label="string"
+				id="created-by-ai">
+				{this.isIconOnly ?
 					<Button id={linkId} icon="sap-icon://ai" design="Transparent" onClick={this._handleToggleClick}></Button> :
-					<Link
-						id={linkId}
-						accessibleRole="Button"
-						onClick={this._handleToggleClick}
-						icon={this.isEmphasized ? "ai" : ""}
-					>
-						{this.attributionText}
-					</Link>
+					this.isEmphasized ?
+						<Link
+							id={linkId}
+							accessibleRole="Button"
+							onClick={this._handleToggleClick}
+							icon="ai"
+						>
+							{this._attributionText}
+						</Link> :
+						<Link
+							id={linkId}
+							accessibleRole="Button"
+							onClick={this._handleToggleClick}
+						>
+							{this._attributionText}
+						</Link>
 				}
 
-				<Label>
-					{!this.isShortened && !this.isIconOnly && this.verificationText}
+				<Label for="created-by-ai">
+					{!this.isShortened && !this.isIconOnly && this._verificationText}
```

Comment:
I missed this in the previous review - right now you render a label with empty content even if we are in shortened mode. Rewrite this part with the check to not render a Label component at all if the mode is iconOnly or Shortened.